app/Nova/Actions/Auth0/BlockAuth0Client.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Nova\Actions\Auth0;
+
+use App\Auth0\Events\ClientBlocked;
+use App\Auth0\Models\Auth0ClientModel;
+use Illuminate\Bus\Queueable;
+use Illuminate\Queue\InteractsWithQueue;
+use Laravel\Nova\Actions\Action;
+use Laravel\Nova\Actions\ActionModelCollection;
+use Laravel\Nova\Fields\ActionFields;
+use Ramsey\Uuid\Uuid;
+
+final class BlockAuth0Client extends Action
+{
+    use InteractsWithQueue;
+    use Queueable;
+
+    public function handle(ActionFields $fields, ActionModelCollection $actionModelCollection): void
+    {
+        foreach ($actionModelCollection as $auth0ClientModel) {
+            if (!$auth0ClientModel instanceof Auth0ClientModel) {
+                continue;
+            }
+
+            ClientBlocked::dispatch(Uuid::fromString($auth0ClientModel->id));
```

Comment:
This should become a [Laravel Job](https://laravel.com/docs/10.x/queues#class-structure) instead of an event. The actual block of the client did not happen, this is just the intention from the user.

So the flow is:
- Create a Job called `BlockClient` with a handle method
- Dispatch the `BlockClient` job here
- Inside the handle method of `BlockClient` the actual event `ClientBlocked` can be dispatched

For now we will not listen to `ClientBlocked` but it can be used in the future to update for example the internal state of a client